src/domManipulation.ts

```diff
@@ -68,16 +79,41 @@ function temperatureColourConverter(temp) {
   }
 }
 
-export function displayForecastLoader(location) {
+export function displayForecastLoader(location: string) {
   if (location === "custom") {
-    document.getElementById("popup-loading-container").style.visibility = "visible";
-    document.getElementById("popup-loading-container").style.display = "flex";
-    document.getElementById("popup-forecast-container").style.visibility = "hidden";
-    document.getElementById("popup-forecast-container").style.display = "none";
+
+    const popupLoader = document.getElementById("popup-loading-container");
+
+    if (popupLoader !== null) {
+      popupLoader.style.visibility = "visible";
+      popupLoader.style.display = "flex";
+    }
+    
+    const popupForecastDisplay = document.getElementById("popup-forecast-container");
+
+    if (popupForecastDisplay !== null) {
+      popupForecastDisplay.style.visibility = "hidden";
+      popupForecastDisplay.style.display = "none";
+    }
+    
   }
 }
 
-export function loadWeatherForecast(location, fetchResults) {
+interface Temperature {
```

Comment:
These can be defined elsewhere if you want in a `*.d.ts` file, usually in a `types` or `models` directory